src/main/java/org/apache/maven/plugins/help/EffectivePomMojo.java

```diff
@@ -111,6 +112,7 @@ public void execute()
         if ( StringUtils.isNotEmpty( artifact ) )
         {
             project = getMavenProject( artifact );
+            projects = Collections.emptyList();
```

Comment:
I am confused why you do this? What else is injected?

Reply:
Maven injects the list of poms (local and sub modules).
Run the included integration test without my change to see what happens.

Reply:
The shouldWriteAllEffectivePOMsInReactor() method checks if there is more than 1 item in projects and returns true even though the artifact argument has been provided which should override this.

Reply:
I'm confused as well... cause all integration tests are running fine https://builds.apache.org/view/M-R/view/Maven/job/maven-box/job/maven-help-plugin/job/master/ ?

Reply:
Those executions do not contain the new test included in my pull request.
Please run that test without my fix to see why it is needed.

Reply:
If you run "mvn help:effective-pom -Dartifact=org.apache.maven.plugins:maven-help-plugin" in a folder with a simple pom (no modules defined in the pom) it will output the effective pom for the maven-help-plugin as expected.
But if you run the same command in a folder with a more complex pom (where there are one or more modules defined in the pom) it will output the effective pom for these insted of the expected output.
This is caused by "artifact" only overriding the "project" variable and not the "projects" variable.
Thinking about it again now I should probably set "projects" to be a list only containing the "artifact". Will update the commit.